src/components/DevPanel.tsx

```diff
@@ -0,0 +1,369 @@
+    sort: true,
+  }, {
+    text: 'Actions',
+    formatter: this.ButtonFormatter,
+  }];
+
+  constructor(props: Props) {
+    super(props);
+    this.handleChangeFileUpload = this.handleChangeFileUpload.bind(this);
+    this.handleFileChange = this.handleFileChange.bind(this);
+    this.handleChangeFileDownload = this.handleChangeFileDownload.bind(this);
+    this.handleFileDownload = this.handleFileDownload.bind(this);
+    this.state = {
+      pdfFiles: undefined,
+      buttonState: '',
+      currentApplicationId: undefined,
+      currentApplicationFilename: undefined,
+      currentUser: undefined,
+      currentPage: 0,
+      itemsPerPageSelected: listOptions[0],
+      numElements: 0,
+      username: props.username,
+      searchName: '',
+      adminName: props.name,
+      organization: props.organization,
+      documents: [],
+    };
+    this.onClickWorker = this.onClickWorker.bind(this);
+    this.handleChangeSearchName = this.handleChangeSearchName.bind(this);
+    this.handleChangeItemsPerPage = this.handleChangeItemsPerPage.bind(this);
+    this.changeCurrentPage = this.changeCurrentPage.bind(this);
+    this.getDocuments = this.getDocuments.bind(this);
+    this.onChangeViewPermission = this.onChangeViewPermission.bind(this);
+    this.ButtonFormatter = this.ButtonFormatter.bind(this);
+  }
+
+  componentDidMount() {
+    this.getDocuments();
+  }
+
+  onClickWorker(event: any) {
+    this.setState({ currentUser: event });
+  }
+
+  handleChangeSearchName(event: any) {
+    this.setState({
+      searchName: event.target.value,
+      currentPage: 0,
+    });
+  }
+
+  handleChangeItemsPerPage(itemsPerPageSelected: any) {
+    this.setState({
+      currentPage: 0,
+    });
+  }
+
+  handleViewDocument(event: any, rowIndex: number) {
+    const {
+      documents,
+    } = this.state;
+
+    const index = rowIndex;
+    const form = documents[index];
+    const {
+      id,
+      filename,
+    } = form;
+    this.setState(
+      {
+        currentApplicationId: id,
+        currentApplicationFilename: filename,
+      },
+    );
+  }
+
+  changeCurrentPage(newCurrentPage: number) {
+    this.setState({ currentPage: newCurrentPage }, this.getDocuments);
+  }
+
+  onChangeViewPermission(event: any) {
+    const {
+      currentUser,
+    } = this.state;
+    currentUser.viewPermission = event.target.ischecked;
+    this.setState({ currentUser }, this.getDocuments);
+  }
+
+  getDocuments() {
+    const {
+      searchName,
+      currentPage,
+      itemsPerPageSelected,
+    } = this.state;
+    const itemsPerPage = Number(itemsPerPageSelected.value);
+    fetch(`${getServerURL()}/get-documents `, {
+      method: 'POST',
+      credentials: 'include',
+      body: JSON.stringify({
+        pdfType: PDFType.FORM,
+        annotated: false,
+      }),
+    }).then((response) => response.json())
+      .then((responseJSON) => {
+        console.log(responseJSON);
+        const {
+          documents,
```

Comment:
we should check the status of the response rather than assuming that there are documents